python/tskit/tables.py

```diff
@@ -1815,3 +1817,23 @@ def unpack_strings(packed, offset, encoding="utf8"):
     :rtype: list[str]
     """
     return [b.decode(encoding) for b in unpack_bytes(packed, offset)]
+
+
+def to_np_int32(int_array):
+    """
+    A few functions require arrays of type np.int32. To allow passing standard numpy
+    integer arrays (dtype=np.int64) we cast but check bounds to avoid wrap-around
+    conversion errors (numpy doesn't seem to provide this functionality)
+    """
+    int_array = np.array(int_array)
+    if int_array.size == 0:
+        return int_array.astype(np.int32)  # Allow empty arrays of any type
+    try:
+        return int_array.astype(np.int32, casting='safe')
+    except TypeError:
+        int32bounds = np.iinfo(np.int32)
+        if (int_array >= int32bounds.min).all() and (int_array <= int32bounds.max).all():
+            # This should raise the correct error on conversion from e.g. a float array
```

Comment:
This comment is a bit unsettling - does it correct for them or not?

Reply:
Yes, `np.array([1.0]).astype(np.int32, casting='same_kind')` raises 

```
TypeError: Cannot cast array from dtype('float64') to dtype('int32') according to the rule 'same_kind'
```

Reply:
As confirmed in the test suite

Reply:
I'm sure it's correct, it's just the comment implies that this hasn't been verified. In a few years someone will read through (probably me) and think that this function hasn't been tested and then waste a bunch of time retesting it. I'm just asking to change the comment so that it doesn't mislead like this.

Reply:
Yes, fine, please change

Reply:
OK, I'll come back and fix the comment when I get a chance then